presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/ColumnReference.java

```diff
@@ -0,0 +1,93 @@
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.metadata.TableHandle;
+import com.facebook.presto.metadata.TableMetadata;
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.TableScanNode;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class ColumnReference
+    implements RvalueMatcher
+{
+    private final String tableName;
+    private final String columnName;
+
+    public ColumnReference(String tableName, String columnName)
+    {
+        this.tableName = requireNonNull(tableName, "tableName is null");
+        this.columnName = requireNonNull(columnName, "columnName is null");
+    }
+
+    @Override
+    public Optional<Symbol> getAssignedSymbol(PlanNode node, Session session, Metadata metadata, ExpressionAliases expressionAliases)
+    {
+        if (!(node instanceof TableScanNode)) {
+            return Optional.empty();
+        }
+
+        TableScanNode tableScanNode = (TableScanNode) node;
+        TableMetadata tableMetadata = metadata.getTableMetadata(session, tableScanNode.getTable());
+        String actualTableName = tableMetadata.getTable().getTableName();
+
+        // Wrong table -> doesn't match.
+        if (!tableName.equalsIgnoreCase(actualTableName)) {
+            return Optional.empty();
+        }
+
+        Optional<ColumnHandle> columnHandle = getColumnHandle(tableScanNode.getTable(), session, metadata);
+
+        checkState(columnHandle.isPresent(), format("Table %s doesn't have column %s. Typo in test?", tableName, columnName));
+
+        return getAssignedSymbol(tableScanNode, columnHandle.get());
+    }
+
+    private Optional<Symbol> getAssignedSymbol(TableScanNode tableScanNode, ColumnHandle columnHandle)
+    {
+        Optional<Symbol> result = Optional.empty();
+        for (Map.Entry<Symbol, ColumnHandle> entry : tableScanNode.getAssignments().entrySet()) {
+            if (entry.getValue().equals(columnHandle)) {
+                checkState(!result.isPresent(), "Multiple ColumnHandles found for %s:%s in table scan assignments", tableName, columnName);
+                result = Optional.of(entry.getKey());
+            }
+        }
+        return result;
+    }
+
+    private Optional<ColumnHandle> getColumnHandle(TableHandle tableHandle, Session session, Metadata metadata)
+    {
+        return metadata.getColumnHandles(session, tableHandle).entrySet()
+                .stream()
+                .filter(entry -> columnName.equals(entry.getKey()))
+                .map(Map.Entry::getValue)
+                .findFirst();
+    }
+
+    @Override
+    public String toString()
+    {
+        return format("%s:%s", tableName, columnName);
```

Comment:
toStringHelper makes the output harder to read. I'll switch it to Column %s:%s, though I'd say it's pretty clear from context.